metricflow/dataflow/builder/dataflow_plan_builder.py

```diff
@@ -309,6 +320,48 @@ def _select_source_nodes_with_meausres(
                 nodes.append(source_node)
         return nodes
 
+    def _find_non_additive_dimension_in_linkable_specs(
+        self,
+        agg_time_dimension: TimeDimensionReference,
+        linkable_specs: Sequence[LinkableInstanceSpec],
+        non_additive_dimension_spec: NonAdditiveDimensionSpec,
+    ) -> Optional[TimeDimensionSpec]:
+        """Finds the TimeDimensionSpec matching the non_additive_dimension_spec, if any"""
+        queried_time_dimension_spec: Optional[LinkableInstanceSpec] = None
+        for linkable_spec in linkable_specs:
+            dimension_name_match = linkable_spec.element_name == non_additive_dimension_spec.name
+            metric_time_match = (
+                non_additive_dimension_spec.name == agg_time_dimension.element_name
+                and linkable_spec.element_name == self._metric_time_dimension_reference.element_name
+            )
+            if dimension_name_match or metric_time_match:
```

Comment:
Are there issues if the user queries for `metric_time__week` and `metric_time__month`? Or is this validated somewhere?

Reply:
They would be allowed to do that

Reply:
We are validating that in here https://github.com/transform-data/metricflow/pull/259